### PR TITLE
fix Tests for extending simple option by setting trivial sys.argv in the tests

### PR DESCRIPTION
**Pull request by JensTimmerman**
really fixing the tests this time.
@itkovian please confirm?

### CODE REVIEW

File: test/script_tools.py [hunk 1]
```diff
@@ -43,6 +41,16 @@ class TestExtendedSimpleOption(TestCase):
     """
     Tests for the ExtendedSimpleOption class.
     """
+    def setUp(self):
+        """Backup sys.argv"""
+        super(TestCase, self).setUp()
+        self._old_argv = sys.argv
+        sys.argv = sys.argv[:1]
```
Comment (stdweird) [inline test/script_tools.py]: why?

File: test/script_tools.py [hunk 2]
```diff
@@ -52,7 +60,7 @@ def test_threshold_default_setting(self, mock_proceed, mock_lock, mock_lockfile)
         mock_proceed.return_value = True
         mock_lockfile.return_value = mock.MagicMock()
 
-        opts = ExtendedSimpleOption(options={}, go_args=[])
```
Comment (stdweird) [inline test/script_tools.py]: why not mock sys.argv?

File: test/script_tools.py [hunk 3]
```diff
@@ -43,6 +41,16 @@ class TestExtendedSimpleOption(TestCase):
     """
     Tests for the ExtendedSimpleOption class.
     """
+    def setUp(self):
+        """Backup sys.argv"""
+        super(TestCase, self).setUp()
+        self._old_argv = sys.argv
```
Comment (stdweird) [inline test/script_tools.py]: should this be a copy?

File: test/script_tools.py (hunk 2 above)
Comment (stdweird) [inline test/script_tools.py]: also, with `**kwargs` fix above, is this still needed?

File: test/script_tools.py (hunk 1 above)
Comment (JensTimmerman) [inline test/script_tools.py]: this really fixes the tests this time (see pr title)
If you run the tests with `python setup.py test --xmlrunner=xxx` later on the tests for ExtendedSimpleOption will fail with `No such option --xmlrunner` 
see https://jenkins1.ugent.be/job/vsc-utils/PYTHON=CPython-2.7/11/console

File: test/script_tools.py (hunk 2 above)
Comment (JensTimmerman) [inline test/script_tools.py]: this was added in error, did not work or fix the problem, so I reverted it again.
see repeated failed attempts to fix the tests https://github.com/hpcugent/vsc-utils/pull/32
 and https://github.com/hpcugent/vsc-utils/pull/33

File: test/script_tools.py (hunk 3 above)
Comment (JensTimmerman) [inline test/script_tools.py]: no
```>>> import sys
>>> sys.argv
['']
>>> a = sys.argv
>>> a
['']
>>> sys.argv = ['foo']
>>> a
['']
>>> sys.argv = a
>>> sys.argv
['']
>>> a
['']
>>> sys.argv = ['foo']
>>> a = sys.argv
>>> sys.argv = ['']
>>> a
['foo']
>>> sys.argv= a
>>> sys.argv
['foo']
```

Comment (JensTimmerman) [inline test/script_tools.py]: no, the reference tot sys.argv is overwritten in the next line. (the [1:] is a copy there.)
But I can make it a bit more explicit for readability if this is confusing
```
>>> import sys
>>> sys.argv = ['foo']
>>> a = sys.argv
>>> sys.argv = sys.argv[:1]
>>> a
['foo']
>>> sys.argv
['foo']
>>> sys.argv.append('bar')
>>> a
['foo']
>>> sys.argv
['foo', 'bar']
>>> 
```

**Comment by itkovian** [comment]
Nice. Thx.

**Pull request merged**